src/main/java/ch/njol/skript/effects/EffOpenInventory.java

```diff
@@ -34,125 +36,209 @@
-			invType = CHEST;
-		} else if (parseResult.mark >= 0) {
-			invType = WORKBENCH;
-			openFlag = parseResult.mark ^ 0;
-		} else {
-			openFlag = parseResult.mark;
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		inventory = exprs.length > 1 ? exprs[0] : null;
+		if (matchedPattern == 1) {
+			open = true;
+			if (!parseResult.tags.isEmpty()) { // %-inventory% was not used
+				syntax = OpenableInventorySyntax.valueOf(parseResult.tags.get(0).toUpperCase(Locale.ENGLISH));
+				if (syntax.getVersion() != null && !Skript.isRunningMinecraft(syntax.getVersion())) {
+					Skript.error("Opening an inventory of type '" + syntax.toString().toLowerCase(Locale.ENGLISH) + "' is only present on Minecraft version " + syntax.getVersion());
+					return false;
+				}
+				if (!syntax.doesMethodExist()) {
+					Skript.error(syntax.getMethodError());
+					return false;
+				}
+			}
 		}
-		
-		open = matchedPattern == 0;
-		invi = open ? exprs[0] : null;
 		players = (Expression<Player>) exprs[exprs.length - 1];
-		if (openFlag == 1 && invi != null) {
-			Skript.warning("Using 'show' inventory instead of 'open' is not recommended as it will eventually show an unmodifiable view of the inventory in the future.");
+		if (inventory instanceof Literal && inventory != null) {
+			Literal<?> literal = (Literal<?>) inventory;
+			Object object = literal.getSingle();
+			if (object instanceof InventoryType && !((InventoryType) object).isCreatable()) {
+				Skript.error("You can't open a '" + literal.toString() + "' inventory to players. It's not creatable.");
+				return false;
+			}
 		}
 		return true;
 	}
-	
+
 	@Override
-	protected void execute(final Event e) {
-		if (invi != null) {
-			Inventory i;
-			
-			assert invi != null;
-			Object o = invi.getSingle(e);
-			if (o instanceof Inventory) {
-				i = (Inventory) o;
-			} else if (o instanceof InventoryType) {
-				i = Bukkit.createInventory(null, (InventoryType) o);
-			} else {
+	protected void execute(Event event) {
+		if (inventory != null) { // Show
+			Inventory inventory = null;
+			Object object = this.inventory.getSingle(event);
+			if (object == null)
 				return;
-			}
-			
-			if (i == null)
-				return;
-			for (final Player p : players.getArray(e)) {
+			if (object instanceof Inventory) {
+				inventory = (Inventory) object;
+			} else if (object instanceof InventoryType) {
 				try {
-					p.openInventory(i);
-				} catch (IllegalArgumentException ex){
-					Skript.error("You can't open a " + i.getType().name().toLowerCase(Locale.ENGLISH).replaceAll("_", "") + " inventory to a player.");
+					inventory = Bukkit.createInventory(null, (InventoryType) object);
+				} catch (Exception e) {
+					// Spigot forgot to label some InventoryType's as non creatable in some versions < 1.19.4
+					// So this throws NullPointerException aswell ontop of the IllegalArgumentException.
```

Comment:
If we know it's going to be an NPE then should we just catch NPE here?
We don't want to give the wrong warning if it's some kind of unexpected internal error.